tests/providers/sftp/triggers/test_sftp.py

```diff
@@ -176,12 +176,11 @@ async def test_sftp_trigger_run_trigger_failure_state(self, mock_get_files_by_pa
         trigger = SFTPTrigger(path="test/path/", sftp_conn_id="sftp_default", file_pattern="my_test_file")
 
         expected_event = {"status": "failure", "message": "An unexpected exception"}
-
-        with pytest.raises(Exception):
```

Comment:
@utkarsharma2 @pankajastro I think this test for verifying the exception is not handled correctly. I hope the with for pytest.raises exception was written to handle the exception thrown by trigger.run(). But right now the code throws exception in the assert part, as the assert didn't match and as its wrapped inside with statement, the test case passes.
It is originally intended to catch the trigger.run() exception right? I will try fixing this part.

Reply:
<img width="1356" alt="Screenshot 2024-03-28 at 11 03 11 AM" src="https://github.com/apache/airflow/assets/10162465/7755d85a-c527-4bfa-a470-f0cf54a5f4f1">
When wrapper trigger.run() part and await part in code inside the with statement, it didn't raise any exception as the code is handled in such way that we don't throw exception but return the error message

Reply:
Yea. It's cool you found it.